addons/sale_project/models/sale_order_line.py

```diff
@@ -125,13 +125,13 @@ def _compute_qty_delivered(self):
             sol_id = line.id or line._origin.id
             line.qty_delivered = reached_milestones_per_sol.get(sol_id, 0.0) * line.product_uom_qty
 
-    @api.depends('order_id.partner_id', 'product_id', 'order_id.project_id')
+    @api.depends('order_id.partner_id', 'company_id', 'product_id', 'order_id.project_id')
```

Comment:
I'm not sure we can force the override of the analytic distribution of the SOL if its company changes, as here https://github.com/odoo/odoo/pull/190687/files#diff-54515e6934b8a75befb26b76cc09b04f898281f6fcfc3f32ac4b1afdd17004b2L132, we will only write the distribution if it does not exist yet :)

The reason we do that is because we have no way to determine if an existing distribution was computed automatically from the product of the SOL or from a distribution model or even manually written by the user for instance. So we cannot really replace the distribution by another if it already set.

Reply:
@deneuvillem Yes I known we're not going to recompute it if there was one already set (same behavior for invoice etc...) but at least it's bit less wrong; maybe should I also mention it on the `_onchange_company_id_warning()` ? (don't it introducing new string is ok regarding the stable policy)

Reply:
@xavieralt Sounds fair :) I guess we could indeed mention it in the warning, so it appears more transparent to the user that changing the SO company could affect some SOLs's distributions.